src/BalanceTx.purs

```diff
@@ -403,11 +405,18 @@ balanceTx unattachedTx@(UnattachedUnbalancedTx { unbalancedTx: t }) = do
     -- Get own wallet address, collateral and utxo set:
     ownAddr <- ExceptT $ getWalletAddress <#>
       note (GetWalletAddressError' CouldNotGetNamiWalletAddress)
-    collateral <- ExceptT $ getWalletCollateral <#>
-      note (GetWalletCollateralError' CouldNotGetNamiCollateral)
+    wallet <- asks _.wallet
     utxos <- ExceptT $ utxosAt ownAddr <#>
       (note (UtxosAtError' CouldNotGetUtxos) >>> map unwrap)
-
+    collateral <- case wallet of
+      Just (Nami _) -> Just <$>
+        ( ExceptT $ getWalletCollateral <#>
+            note (GetWalletCollateralError' CouldNotGetNamiCollateral)
+        )
+      -- TODO(jy14898) Supply with fee estimate?
```

Comment:
Please open issues corresponding to the TODOs (and add their URLs in the comment)